Source/WebCore/html/HTMLInputElement.cpp

```diff
@@ -1069,7 +1072,7 @@ bool HTMLInputElement::supportsWritingSuggestions() const
         InputType::Type::Email,
     };
 
-    return allowedTypes.contains(m_inputType->type());
+    return allowedTypes.contains(protectedInputType()->type());
```

Comment:
We have to be careful and ideally such PR would be guided by static analysis. The previous code was safe since `type()` is inline and trivial. The static analysis would not warn about this and the new code is a bit slower unnecessarily. There may be other cases of this in this PR but it's hard for me to review all of them (I thought this was guided by static analysis)